noxfile.py

```diff
@@ -1,9 +1,47 @@
+from pathlib import Path
+
 import nox
 
-@nox.session(python=["3.11"]) # The python version should match the readthedocs configuration.
+requirements_files = list(
+    {path.stem for path in Path.cwd().glob("**/*.in")}
+)
```

Comment:
Looking into this more, I think it's wasteful to have this script inspect arbitrary subfolders in the project. Scanning the file system like this often ends up having an undesired overhead. Besides, it's been decided to consolidate the requirements in a predictable location.
Additionally, it's useful to calculate the folder path relative to the current script, regardless from the current working directory.
Finally, having that path in a variable/constant, makes it reusable across the module, when it's desired.
```suggestion
requirements_directory = (Path(__file__).parent / "requirements").resolve()

requirements_files = [
    requirements_input_file_path.stem
    for requirements_input_file_path in requirements_directory.glob("requirements/*.in")
]
```

Reply:
The script always runs with CWD set to the repository root, so you don't need the whole `Path(__file__)` part. Just use `Path("requirements").glob(...)` directly.

Reply:
Thanks again @webknjaz and @gotmax23 I really like the suggestion of not scanning arbitrary folders as well as using the path as a variable. I've pushed some commits.